### ISSUE TITLE
Add metadata to search & query analytics

### ISSUE THREAD

**Issue by samsha1** [typesense/typesense#1972]
## Description

Hi, I was trying search & query analytics with Typesense and I was wondering if there is a way to send a metadata to queries collection so that we know using what filter the system did/didn't return any result . is there a way to send such metadata in queries & no_hits_queries collection?

Currently, We've query count and query name for search analytics but we miss the metadata / search parameter and it's not clear using what filter the search result did/didn't return the data.

## Expected Behavior

It would be better to store request parameters (i.e. filter rules).

## Actual Behavior

Only query and count are stored which makes very hard to analyze without filter

## Metadata

**Typesense Version**: 26.*, 27.*

**Comment by kishorenc** [comment]
We don't have a way to associated metadata. If we did that, then aggregation of counts need to happen at a `query + filter_rules`  level. This can potentially be a very high cardinality aggregation. At that point, allowing a simple logging + export of such requests is a better idea than allowing count aggregation?

**Comment by kishorenc** [comment]
Yeah it will just be a running log which you can export and then process for aggregation as you see fit.
